### ISSUE TITLE
Add an option to log all inputs and outputs to timestamped per-tab / per-session files.

### ISSUE THREAD

**Issue by seantcanavan** [warpdotdev/Warp#4833]
### Describe the solution you'd like?

It would be awesome if warp could output all inputs and outputs per tab / session to a timestamped file. maybe even with the tab index as well?

### Is your feature request related to a problem? Please describe.

I was doing a lot of infra / docker debugging on my local and while the solution was eventually committed to github, the root cause wasn't always obvious when debugging complex systems.

There were multiple times in the last week where I really wish Warp could write all inputs and all outputs per-tab / per-session to a uniquely named file that way I could go back in time and search for error messages I ended up fixing. 

`.zhistory` is basically one quarter of this new feature request but isn't unique per session and is only the inputs.

### Additional context

chatGPT spit out an interesting attempt at using `script` to achieve this functionality:

```
function start_logging_session() {
    if [[ -z "$SCRIPT_LOGGING_STARTED" ]]; then
        export SCRIPT_LOGGING_STARTED=1
        local log_dir="$HOME/terminal_logs"
        mkdir -p "$log_dir"
        local log_file="${log_dir}/session_$(date +%Y%m%d_%H%M%S)_$$.txt"
        script "$log_file"
    fi
}
```

but this has its own problems. for instance, without the `export` line you create an infinite loop of infinite shells and also infinite log outputs. and then you very quickly cascade disk space as you're running up to hundreds/thousands of shells and writing every terminal command to all of them. i accidentally used about 300MB of disk space in just a few seconds of terminal use. 

finally, if you `cat` one of the hundreds of output files you create another kind of infinite loop where you're essentially cat'ing a file that's being written to in real time and depending on disk i/o it could go on forever as the cat output gets written right back to the file you're `cat`'ing.

### How important is this feature to you?

3

### Warp Internal (ignore) - linear-label:39cc6478-1249-4ee7-950b-c428edfeecd1

None

**Comment by dannyneira** [comment]
Thanks for this feature request! 

To anyone else interested in this feature, please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

**Comment by dannyneira** [comment]
Hey folks, closing in favor of https://github.com/warpdotdev/Warp/issues/2394

**Comment by dannyneira** [comment]
Duplicate of https://github.com/warpdotdev/Warp/issues/2394

**Comment by RobBednark** [comment]
I would love to see this feature.  I've been using it for years in iTerm2 for all my sessions, and it's sometimes invaluable to go back and see where I've done something hours/days/months ago.  For reference, the iTerm2 docs, and a screenshot of my setting:

---
[iTerm2 Documentation](https://iterm2.com/documentation-one-page.html)
**Automatically log session input to files in:**
If enabled, every session's output will be logged to a file in the specified directory. The filename format can also be specified here as an interpolated string.

**Log format**
You can log in the following formats:

_Raw data_ - An exact copy of the input that was received including control sequences.
_Plain text_ - Excludes control sequences. Just text and newlines. This is very readable when running CLI apps but becomes utter chaos for interactive programs like vim or emacs.
_HTML_ - Like Plain text, it excludes control sequences. Colors and various font attributes are preserved. The resulting file can be viewed in a web browser.
_asciinema_ - Produces a file viewable with [asciinema](https://asciinema.org/), which preserves time as well as more font attributes.

---
**iTerm2 settings:** 

<img width="1030" alt="Image" src="https://github.com/user-attachments/assets/7c4da20d-735a-4615-ab54-962783b9dc1f" />

**Comment by endersuu** [comment]
This feature is essential for me to review the commands I’ve entered a few days ago when reproducing bugs.

**Comment by kmanc** [comment]
This is the feature that stands between me and making Warp my daily driver. I may be unique in that regard but it does seem like a common request

**Comment by michaeljgoodman** [comment]
Similar to others, the absence of this feature stops this from being my daily driver.

**Comment by AlefCS** [comment]
Does this include the possibility to dump the whole terminal session to a file like [this Terminator plugin](https://gnome-terminator.readthedocs.io/en/latest/plugins.html#logging-dump) or should I open a new issue?

I'm not interested in every session to be logged but sometimes I want to manually save them.

**Comment by michaeljgoodman** [comment]
Yes, similar desire from me. I don't necessarily desire all my sessions being logged out but I do want to be able to export a whole session by choice for audit purposes.

And similarly the lack of this feature is my primary reason for not using Warp right now.

**Comment by ChrisNelsonOK** [comment]
Very much needed, kind of crazy to _not_ have this added yet.
